app/components/global_threads/thread_options/index.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+import {bindActionCreators, Dispatch} from 'redux';
+
+import {showPermalink} from '@actions/views/permalink';
+import {
+    flagPost,
+    setUnreadPost,
+    unflagPost,
+} from '@mm-redux/actions/posts';
+import {setThreadFollow, updateThreadRead} from '@mm-redux/actions/threads';
+import {getCurrentUserId} from '@mm-redux/selectors/entities/common';
+import {getPost} from '@mm-redux/selectors/entities/posts';
+import {getMyPreferences, getTheme} from '@mm-redux/selectors/entities/preferences';
+import {getCurrentTeam, getCurrentTeamUrl} from '@mm-redux/selectors/entities/teams';
+import {getThread} from '@mm-redux/selectors/entities/threads';
+import {isPostFlagged} from '@mm-redux/utils/post_utils';
+import {getDimensions} from '@selectors/device';
+
+import ThreadOptions, {OwnProps} from './thread_options';
+
+import type {GlobalState} from '@mm-redux/types/store';
+
+export function makeMapStateToProps() {
+    return (state: GlobalState, ownProps: OwnProps) => {
+        const myPreferences = getMyPreferences(state);
+        return {
+            ...getDimensions(state),
```

Comment:
do you need this? can you just use the Dimensions module in the component ?

Reply:
We are using same code as `post_options`. I thought it must have been written for the responsiveness (Landscape to portrait or vice-versa)

Reply:
yes, responsiveness, but we can just stop using the redux values, that was all. Anyway does not apply anymore